### ISSUE TITLE
[GAME BREAKING] Certain config setups cause upgrading to be impossible

### ISSUE THREAD

**Issue by ImpulsiveLass** [HennyBicks01/LethalProgressionPlus#1]
https://streamable.com/rrzhdr
I have never had the +1 button work but now that I removed a 6th stat and its down to the five, none of the 3 buttons to level up work.

**Comment by h-nny** [comment]
weird ill look into it now

**Comment by ImpulsiveLass** [comment]
Thank you, I'll quickly go test it alone to make sure its not a compat issue

**Comment by ImpulsiveLass** [comment]
This is with only the mod itself, bepinex, and lc_api installed
https://streamable.com/jb8wn7
The bug seems to be config related yeah

**Comment by h-nny** [comment]
thats kinda what i figured since if it was a compatibility issue it would just not work at all

**Comment by h-nny** [comment]
have you tried playing with unspec enabled???

**Comment by ImpulsiveLass** [comment]
Yes, in ship only. +1 still didn't work, i'll test again with the rest of the current config

**Comment by ImpulsiveLass** [comment]
Tested it
https://streamable.com/q0dcu2
Doesn't work still
Interesting note is that all the plus things vacuums JMP to be selected and all the subtraction vacuums SPD to be selected

**Comment by h-nny** [comment]
what version are you playing on??? lethalprogressionplus???

**Comment by ImpulsiveLass** [comment]
Yes, most recent release

**Comment by h-nny** [comment]
ok ill look into what could be causing that

**Comment by h-nny** [comment]
try downloading this one and uninstalling the other one https://thunderstore.io/c/lethal-company/p/Henny/RogueCompanyPlus/

**Comment by h-nny** [comment]
its the same mod under a different name since the original creator didn't like the name of it

**Comment by ImpulsiveLass** [comment]
Using roguecompanyplus, with no config changes its working correctly
https://streamable.com/0j1egv

**Comment by ImpulsiveLass** [comment]
Hmmm very interesting, the issue does persist
+5, -5, +max, -max work as intended. +1 changes it to speed, and -1 changes it to vision now (also vision is there regardless of whether its checked true or false)
https://streamable.com/isvmix

**Comment by ImpulsiveLass** [comment]
Ahh, I just noticed that when you hover over +1 and -1 you can see the red outline on those respective stats
Also you may notice in the clip once or twice the +5 broke and sent me to a different stat

**Comment by ImpulsiveLass** [comment]
Ig the mapping that the buttons use are truncated when skills inbetween are taken out and it doesn't link up correctly? I'm not sure I'm amateur with code

**Comment by h-nny** [comment]
no youre almost definitely right ill look into why that might be happening

**Comment by h-nny** [comment]
omg i know whats happening

**Comment by h-nny** [comment]
the hitboxes for the buttons are overlapping because they're are so few the go over the +/- 1

**Comment by h-nny** [comment]
let me put out a fix real quick

**Comment by ImpulsiveLass** [comment]
Cool thanks, I'm going out in a minute so I can test it in a few hours

**Comment by h-nny** [comment]
ok i put out an update that fixes the problem so lmk if anything else comes up

**Comment by ImpulsiveLass** [comment]
I’m still out but did you also fix the vision Boolean?

**Comment by ImpulsiveLass** [comment]
Where can I update? I actually don't see rogue company plus up on thunderstore now or at least I can't find it

**Comment by ImpulsiveLass** [comment]
Hello?

**Comment by h-nny** [comment]
sorry about that it looks like it got taken down again for no reason i went ahead and uploaded a release to this github detailing how you can replace the dll file you have

**Comment by ImpulsiveLass** [comment]
That sucks, I got the dll to work and yes the +/- 1 are working correctly. Now 3 other buggies
1. The vision boolean does not remove it from the game
![Screenshot 2023-12-23 133156](https://github.com/HennyBicks01/LethalProgressionPlus/assets/92990441/0866e944-5455-424b-a516-492b094bb424)
2. Not really an issue but my punk ass friend abuses it, your skills are reset when you leave and rejoin so upspec is kinda always enabled in that way
3. After being ejected the game intially displays that you still have stats leveled up until you try to add/remove lvls from them then it show that it is really 0

**Comment by h-nny** [comment]
for the vision one try to delete the config file and relaunch then edit the new one sometimes when new things get added to old configs R2 doesn't like it. with the other two it's kinda hard to get around the other two I know Stoneman is tying saves to Steam IDs on the next release of Lethal Progression, and for the last one, it might just not be a check to update the UI until after you try to put a point in a skill. so the number is right in the backend but just isn't being displayed correctly upfront until an action is taken

**Comment by ImpulsiveLass** [comment]
Oh so it checks what values are shown whenever you change them? Couldn't a check be added for when the players are ejected and it restarts?

**Comment by ImpulsiveLass** [comment]
I deleted my config and let it regenerate, vision is still applied whether or not its true or false

**Comment by h-nny** [comment]
ok i found what the problem was ill put out another release

**Comment by h-nny** [comment]
also yeah you can but its kind of complicated to find and pin down whether its a unique check or not

**Comment by ImpulsiveLass** [comment]
Yup that pack worked 👍 
![Screenshot 2023-12-23 163313](https://github.com/HennyBicks01/LethalProgressionPlus/assets/92990441/0da44fd6-6a24-47a1-a9d2-a712b003ca63)

**Comment by h-nny** [comment]
Awesome
